internal/lsp/cache/pkg.go

```diff
@@ -65,6 +72,21 @@ func (pkg *Package) GetActionGraph(ctx context.Context, a *analysis.Analyzer) (*
 		pkg.analyses[a] = e
 		pkg.mu.Unlock()
 
+		defer func() {
+			// If we got an error, clear out our defunct cache entry. We don't cache
+			// errors since they could depend on our dependencies, which can change.
+			// Currently the only possible error is context.Canceled, though, which
+			// should also not be cached.
+			if !e.succeeded {
+				pkg.mu.Lock()
+				delete(pkg.analyses, a)
```

Comment:
Innocent bystander with no context here, but this looks racy? What if another request succeeded and put a non-"defunct" entry into `pkg.analyses[a]` before this defer is able to acquire the lock?

Reply:
That shouldn't be possible because any other request for same package and analyzer will read from the `analyses` cache and wait on the `done` channel. Further up in the method you can see we insert into the `analyses` cache first, before we start the actual work.

Reply:
Also FYI if you have comments you should do them on the Gerrit CL. I'm the only one who saw this comment.